### PR TITLE
Refactor service bus bundle

### PR DESCRIPTION
**Pull request by lunetics**
The DI extension configuration now relies of the provided messagebus class given in the xml configuration instead of the hardcoded mapping.

This makes it easier to provide own classes / messagebus implementations (e.g. debug / logging)

Decorate the Command / Event / Query bus to have busname / bustype an…
…d plugin information / configuration

For each configured plugin (through tagged service and bus config), an "addMethodCall('addPlugin') is now added to the bus definition, eliminating the need for factory creation and makes it possible to tag plugins private
Removed the message_bus factory

Added a PSR logger, where each bus definition has its own logger channel
The messagefactory, router and container plugin are now tagged services (plugin)

Added a Symfony Datacollector Bus-Plugin and view to have extended information about the bus configuration and events in the symfony profiler
Added a Stopwatch plugin to get performance metrics on the symfony performance tab

Remove the .class parameter from xml config
Bad practice and error prone:
also see https://github.com/symfony/symfony/issues/11881
and https://github.com/symfony/symfony/pull/9003

### CODE REVIEW

**Comment by dragosprotung** [comment]
Is there any update with this PR? Is help required with something ?

**Comment by codeliner** [comment]
Tests are failing and and review would be nice. @lunetics anything you want to do/add before merge?

**Comment by lunetics** [comment]
Still working a bit on it, you are free too test... currently not alot of time...

**Comment by dragosprotung** [comment]
I am willing to continue work on this, but except for fixing tests I am not sure what else is planned

**Comment by codeliner** [comment]
@lunetics 
> Still working a bit on it

Can you detail this ^^ I'm also interested in getting the PR merged and if @dragosprotung can help here that would be great. You know next week I'm talking about prooph bundles at Symfony UG meetup in Berlin. Would be good to have the features available.

**Comment by sandrokeil** [comment]
Please wait until #27 and #28 are merged and rebase your branch against master.

File: src/DependencyInjection/Compiler/PluginsPass.php
Comment (sstok) [inline src/DependencyInjection/Compiler/PluginsPass.php]: This can be outside of the loop, the $container is reference.

In fact you can also make this a private method and keep the container as property, but as this method is only called once (during container building) this will not make a big difference.

File: src/DependencyInjection/Compiler/PluginsPass.php
Comment (sstok) [inline src/DependencyInjection/Compiler/PluginsPass.php]: Can also live outside the loop.

File: src/DependencyInjection/Compiler/PluginsPass.php
Comment (sstok) [inline src/DependencyInjection/Compiler/PluginsPass.php]: array_keys() never returns null, no need for coalesce-operator.

File: src/DependencyInjection/Compiler/RoutePass.php
Comment (sstok) [inline src/DependencyInjection/Compiler/RoutePass.php]: Can use coalesce operator instead (also works for undefined 👍).

File: src/DependencyInjection/Compiler/RoutePass.php
Comment (sstok) [inline src/DependencyInjection/Compiler/RoutePass.php]: Symfony 3.3 added a method `ContainerBuilder::getReflectionClass()` for reflection that also takes care of resource tracking. https://github.com/symfony/symfony/pull/21419

**Review by sstok — CHANGES_REQUESTED** [review]

**Comment by codeliner** [comment]
I've rebased the branch with latest master, fixed the tests and checked @sstok ' comments.
I've removed a lot of code from `PluginPass` because it is no longer needed (collected plugin config for ServiceBusFactory, which is removed with this PR)

@lunetics I've disabled the stop watch plugin (commented out in service_bus.xml) because otherwise it is always added to each bus. Would be better to find a way to only add the plugin in debug mode.
@dragosprotung If you're still interested, you could take a look.

I merge the PR as-is, so that I can move on and test the bundle in prooph-do-symfony. Stop watch plugin and other stuff can be added with a new PR if needed.

**Comment by sstok** [comment]
@codeliner Thank you for moving this integration forward again 👍

**Pull request merged**
